api-tests/dev_apis/crypto/test_c009/test_data.h

```diff
@@ -52,14 +52,14 @@ static test_data check1[] = {
  PSA_ALG_KEY_AGREEMENT(PSA_ALG_ECDH, PSA_ALG_HKDF(PSA_ALG_SHA_256)),
  "abcdefghijklmnop", 16,
  PSA_KEY_DERIVATION_INPUT_LABEL,
- PSA_SUCCESS,
```

Comment:
In addition to this fix, you need to fix the error code in line 41 from PSA_ERROR_INVALID_ARGUMENT to PSA_SUCCESS. A new test should be added to test PSA_KEY_DERIVATION_INPUT_SALT cannot be invoked after PSA_KEY_DERIVATION_INPUT_SECRET.

Reply:
Hi @soby-mathew ,

API: psa_key_derivation_input_bytes()
Test Case: Calling this function with different step values.

In this test case, we try to pass different step values to psa_key_derivation_input_bytes() after the psa_key_derivation_setup() is done.

psa_key_derivation_input_bytes() does not accept key as input. 
Instead psa_key_derivation_input_key() must be used (stated in https://github.com/ARMmbed/mbed-crypto/blob/psa-crypto-api-1.0-beta3/include/psa/crypto.h#L3036)
	
The description of the step "PSA_KEY_DERIVATION_INPUT_SECRET" is "This must be a key of type PSA_KEY_TYPE_DERIVE"

Hence passing PSA_KEY_DERIVATION_INPUT_SECRET to psa_key_derivation_input_bytes() throws PSA_ERROR_INVALID_ARGUMENT (stated in https://github.com/ARMmbed/mbed-crypto/blob/psa-crypto-api-1.0-beta3/include/psa/crypto.h#L3052)

And the test_c009 expects the same

Regards,
Gowtham

Reply:
Ignore this comment. PSA_KEY_DERIVATION_INPUT_SECRET is an invalid input to psa_key_derivation_input_bytes().